src/OrchardCore/OrchardCore.Data.YesSql/DbConnectionValidator.cs

```diff
@@ -71,6 +75,13 @@ public async Task<DbConnectionValidatorResult> ValidateAsync(string databaseProv
             return DbConnectionValidatorResult.InvalidConnection;
         }
 
+        // If the shell settings come from the same database, the document table may already exist.
+        if (_shellsSettingsSources.GetType().Name == "DatabaseShellsSettingsSources")
```

Comment:
Use `nameof`

Reply:
Yes would have been better.

But no luck `DatabaseShellsSettingsSources` is defined in `OC` and here the project `OC.Data.YesSql` only references `OC.Abstractions` ;)

Otherwise I would have used `if (_shellsSettingsSources is DatabaseShellsSettingsSources)`

Reply:
@jtkech At what point would the table exists before the tenant is setup? I don't know if that check belongs in the database validator. Maybe add this check during setup instead?

Reply:
I see